Add page on Waypointing

F0mx opened the pull request:
Page on waypointing;
- How to access waypoint editor mode
- Waypoint types
- Waypoint editor Actions
- Interaction with `func_nzp_door` & `wayTarget` description
- Pathing behaviours

F0mx committed:
Create waypoints.md

F0mx committed:
Delete markdown/mapping/waypoints.md

F0mx committed:
Create waypointing.md

Add page on Waypointing

F0mx committed:
Add files via upload

Initial Commit

F0mx committed:
Add files via upload

Initial Commit

F0mx committed:
Update waypointing.md

F0mx committed:
Add files via upload

F0mx committed:
Delete markdown/res/images/waypoint_controls.png

F0mx committed:
Delete markdown/res/images/waypoint_controls.webp

F0mx committed:
Add files via upload

F0mx committed:
Update waypointing.md

F0mx committed:
Update waypointing.md

F0mx committed:
Update waypointing.md

F0mx committed:
Update index.md

F0mx committed:
Update index.md

Added waypointing page link

F0mx committed:
Update waypointing.md

MotoLegacy commented on markdown/mapping/waypointing.md:
```diff
@@ -0,0 +1,40 @@
+
+_Nazi Zombies: Portable_ makes use of a waypointing system to define nodes that zombies may path between. This allows for a fairly simple and intuitive, albeit limited, way to allow mappers to create custom pathfinding for their maps.
+
+Waypoints can be edited directly in-game via the waypoint editing mode, foregoing the need to recompile the map when creating or adjusting waypoints. This mode can be accessed via the command `waypoint_mode 1` in the console.
```
Nit:
_Waypoints are edited directly in-game via a special Waypoint Mode, as opposed to creating them in the map editor. This mode can be accessed via `waypoint_mode` cvar._

MotoLegacy commented on markdown/mapping/waypointing.md:
```diff
@@ -0,0 +1,40 @@
+_Nazi Zombies: Portable_ makes use of a waypointing system to define nodes that zombies may path between. This allows for a fairly simple and intuitive, albeit limited, way to allow mappers to create custom pathfinding for their maps.
+
+Waypoints can be edited directly in-game via the waypoint editing mode, foregoing the need to recompile the map when creating or adjusting waypoints. This mode can be accessed via the command `waypoint_mode 1` in the console.
+Saving a set of waypoints via this mode will generate a `.way` file within the maps folder of NZ:P; a plain-text file containing data for each node.
```
Can we have a section breaking down the format?

MotoLegacy commented on markdown/mapping/waypointing.md:
```diff
@@ -0,0 +1,40 @@
+
+Special Waypoint nodes are used in conjunction with a valid `func_nzp_door` to prevent zombies from attempting to path through currently closed doors.
+
+A valid `func_nzp_door` requires an ID to be set in the `wayTarget` entity field.
```
Can you explain how we select a `wayTarget` in Waypoint Mode?

MotoLegacy commented on markdown/mapping/waypointing.md:
```diff
@@ -0,0 +1,40 @@
+
+## Pathing Behaviour
+
+Zombies will move along a path dictated by the mapper through connected waypoint nodes until they are within sight range to path to a player on their own (Where "Sight" is dictated by both the zombies shins and forehead being able to see the player).
```
Shins, chest, and forehead (there are 3 layers of verticality)

MotoLegacy commented on markdown/mapping/waypointing.md:
```diff
@@ -0,0 +1,40 @@
+Zombies will move along a path dictated by the mapper through connected waypoint nodes until they are within sight range to path to a player on their own (Where "Sight" is dictated by both the zombies shins and forehead being able to see the player).
+
+When zombies reacquire a path to the player (Whether that be after hopping a barricade post-spawn, or in attempt to find the player after they have teleported outside and back into the map.), 
+they will do so via the closest waypoint node to them. This is also important to consider in maps with verticality, as the nearest node to the zombie may in fact be above or below
```
`This is also important to consider in maps with verticality, as the nearest node to the zombie may in fact be above or below[...]`
Not true, because we calculate heuristic cost.

MotoLegacy commented on markdown/mapping/waypointing.md:
```diff
@@ -0,0 +1,40 @@
+
+## Introduction
+
+![Zombies pathfinding between waypoint nodes on Nacht Der Untoten.](../res/images/Waypoint_Pathing.webp)
```
Neat gif!

MotoLegacy reviewed (changes requested):
Just small nitpicks and some requests for expansion, thanks

F0mx committed:
Update waypointing.md

- Corrected "`waypoint_mode 1` in the console " to "waypoint_mode cvar"
- Explanation on wayTarget targetting
- Correction for "Zombie sight" conditions
- Correction for "Zombie pathfinding priority"

F0mx committed:
Update waypointing.md

F0mx committed:
Add files via upload

F0mx committed:
Update waypointing.md

- Added explanation of .way file structure

F0mx commented:
- Corrected "`waypoint_mode 1` in the console " to "waypoint_mode cvar"
- Explanation on wayTarget targetting
- Correction for "Zombie sight" conditions
- Correction for "Zombie pathfinding priority"
- Added explanation of .way file structure

MotoLegacy commented on markdown/mapping/waypointing.md:
```diff
@@ -0,0 +1,53 @@
+
+Special Waypoint nodes are used in conjunction with a valid `func_nzp_door` to prevent zombies from attempting to path through currently closed doors.
+
+A valid `func_nzp_door` requires an ID to be set in the `wayTarget` entity field. Targetting this `wayTarget` within waypoint mode is as simple as placing a waypoint node within the door entity you have created, and creating a special entity
```
"is as simple as" -> "is done via"

MotoLegacy commented on markdown/mapping/waypointing.md:
```diff
@@ -0,0 +1,53 @@
+
+_Nazi Zombies: Portable_ makes use of a waypointing system to define nodes that zombies may path between. This allows for a fairly simple and intuitive, albeit limited, way to allow mappers to create custom pathfinding for their maps.
+
+Waypoints can be edited directly in-game via the waypoint editing mode, foregoing the need to recompile the map when creating or adjusting waypoints. This mode can be accessed via the command `waypoint_mode` cvar.
```
Please consider adopting the other correction I had requested to, pertaining to the choice of words in the map editor tect.

MotoLegacy commented on markdown/mapping/waypointing.md:
```diff
@@ -0,0 +1,53 @@
+Waypoints can be edited directly in-game via the waypoint editing mode, foregoing the need to recompile the map when creating or adjusting waypoints. This mode can be accessed via the command `waypoint_mode` cvar.
+Saving a set of waypoints via this mode will generate a `.way` file within the maps folder of NZ:P; a plain-text file containing data for each node.
+
+With `.way` files being plain-text, they can be opened in any text editor. Doing so will reveal the following data structure:
```
The format breakdown should be it's own category, and should actually describe the format instead of re-stating that it is plaintext and having an example image. Let me know if you'd prefer that I do this.

MotoLegacy commented on markdown/mapping/waypointing.md:
```diff
@@ -0,0 +1,53 @@
+| Special | The associated `wayTarget` ID of a valid `nzp_func_door` that has been correctly linked to the waypoint node. | 
+| Target 1-8 | 8 Possible waypoint nodes that are connected to the waypoint node. | 
+
+Another limitation of the waypoint system is that there is a hard limit of 256 total nodes.
```
Please consider changing this to *Up to 256 total nodes are supported at this time.*

MotoLegacy commented on markdown/mapping/waypointing.md:
```diff
@@ -0,0 +1,53 @@
+
+Another limitation of the waypoint system is that there is a hard limit of 256 total nodes.
+
+## Interacting with Waypoints
```
Having an image for the waypoint mode would be handy, but feel free to proceed without

MotoLegacy reviewed (changes requested):
Thanks for your adjustments, gave a further review

F0mx committed:
Update waypointing.md

-wayTarget wording altered to use "is done via".
-Original "Nit" edit remedied.
-Minor .way formatting tweaks, still requires additional descriptions added.
-"256 Nodes" wording altered, also added known limitation / bug information.

F0mx committed:
Add files via upload

F0mx committed:
Update waypointing.md

-Added example image for waypoint mode UI

F0mx commented:
- wayTarget wording altered to use "is done via".
- Original "Nit" edit remedied.
- Minor .way formatting tweaks, still requires additional descriptions added.
- "256 Nodes" wording altered, also added known limitation / bug information.
- Added example image for waypoint mode UI

MotoLegacy commented:
LGTM, thanks!
